src/main/java/com/jvm_bloggers/core/github/ContributorsService.java

```diff
@@ -32,21 +31,16 @@ public ContributorsService(@GithubClient Client client, GithubProperties githubP
     @Cacheable("contributors")
     public List<Contributor> fetchContributors() {
         WebTarget target = client
-            .target("{api_url}/repos/{org}/{repo}/contributors")
+            .target("{api_url}/repos/{org}/{repo}/contributors?per_page={page_size}")
             .resolveTemplate("api_url", properties.getApiUrl(), false)
             .resolveTemplate("org", properties.getOrg(), false)
-            .resolveTemplate("repo", properties.getRepo(), false);
-        return List.ofAll(traversePages(target).collect(Collectors.toList()));
+            .resolveTemplate("repo", properties.getRepo(), false)
+            .resolveTemplate("page_size", properties.getPageSize(), false);
+        return List.ofAll(getContributorsStream(target).collect(Collectors.toList()));
     }
 
-    private Stream<Contributor> traversePages(WebTarget target) {
+    private Stream<Contributor> getContributorsStream(WebTarget target) {
         Response response = target.request().get();
-        Stream<Contributor> currentPage = response.readEntity(CONTRIBUTORS_LIST_TYPE).stream();
-
-        return Option.of(response.getLink("next"))
-            .map(client::target)
-            .map(this::traversePages)
-            .map(nextPage -> Stream.concat(currentPage, nextPage))
-            .getOrElse(() -> currentPage);
+        return response.readEntity(CONTRIBUTORS_LIST_TYPE).stream();
```

Comment:
In the current implementation when we do need to merge multiple streams there is no need for `List`->`Stream`->`List` conversion, co you can easily return `List` here